src/types.jl

```diff
@@ -0,0 +1,65 @@
+#### Types
+
+"""
+    AbstractParameterSet
+
+The top-level super-type parameter set.
+"""
+abstract type AbstractParameterSet end
+
+"""
+    AbstractEarthParameterSet <: AbstractParameterSet
+
+An earth parameter set, specific to planet Earth.
+"""
+abstract type AbstractEarthParameterSet <: AbstractParameterSet end
+
+"""
+    AbstractMicrophysicsParameterSet <: AbstractParameterSet
+
+An earth parameter set, specific to planet Earth.
+"""
+abstract type AbstractMicrophysicsParameterSet <: AbstractParameterSet end
+
+"""
+    AbstractCloudParameterSet  <: AbstractMicrophysicsParameterSet
+
+A cloud parameter set, for atmospheric clouds
+"""
```

Comment:
Should we also say in the description that they consist of a subset of microphysics parameters, etc?

Reply:
The structure of the concrete type is flexible from the perspective of CLIMAParameters, so I don't think we need to include this information.